src/providers/recordedit.tsx

```diff
@@ -104,6 +106,17 @@ export const RecordeditContext = createContext<{
   lastContiguousChunkRef: any,
   /* max rows allowed to add constant */
   MAX_ROWS_TO_ADD: number,
+  /* the prefill object from cookie storage based on prefill query param */
+  prefillObject: PrefillObject | null,
```

Comment:
When you move the prefill/association logic to ermrestjs, you most probably don't need any of these props. You most probably can either get them from the `reference` itself. Or similar to how `citation` works in Record provider, we could just have one prop here which would have all the APIs and props that you need in it.

Also I feel like `setPrefillAssociationSelectedRows` and `updateAssociationSelectedRows` are very similar to each other. If you pass `prefillAssociationSelectedRows` to the foreignkey components, you don't need `updateAssociationSelectedRows` function anymore.

Reply:
I think the calculated `prefillObject` could be moved entirely to something like `reference.prefill.prefillObject` in ermrestJS. I'm going to wait until we discuss the changes and review this chaise PR before making this decision since it will cause more refactoring in the code.

Regarding `updateAssociationSelectedRows`, I didn't want to have to have the contents of this function four times (twice in each of foreign key field and foreign key dropdown field) in case the functionality needed to change.